### ISSUE TITLE
Investigate telemetry possibly not getting in?

### ISSUE THREAD

**Issue by antkmsft** [Azure/azure-sdk-for-cpp#2738]
From @kyle-patterson:

"[...] I can see that for storage up through -beta.9, but there's no new data coming in from libraries since then. And for keyvault, I can only see two instances of "azsdk-cpp-keyvault/7.2 {os info...}" , where I would have expected keyvault-keys (and more data coming through...) [...]

It almost seems like there was a change in core in the june or july release, but I haven't found anything yet that seems related...

I did confirm that keyvault is passing the expected value when creating the ttp pipeline: https://github.com/Azure/azure-sdk-for-cpp/blob/83295c69edc5d2a6594c09fcd06eae7073753171/sdk/keyvault/azure-security-keyvault-keys/src/key_client.cpp#L28

it's not hyper-critical, but I'd like to get it resolved in the next release so that we can start getting data again..."

**Comment by antkmsft** [comment]
I wrote a simple app that invokes key creation API.
In order to avoid setting up HTTPS decryption on my local machine, I modified the SDK to send requests to a different server on the internet instead of Azure, and read headers there. I compiled with both libcurl and WinHTTP transport adapters. Both have `azsdk-cpp-keyvault-keys/7.2 (Windows 10 Enterprise 6.3 19043 19041.1.amd64fre.vb_release.191206-1406)` as `User-Agent` header. (Note that the version does not match the package version, I opened a bug for that - https://github.com/Azure/azure-sdk-for-cpp/issues/2765). But other than that, it look like everything works as it supposed to.
